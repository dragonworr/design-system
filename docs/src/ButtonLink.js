import { Button } from 'pcln-design-system'
export default Button.withComponent('a')
