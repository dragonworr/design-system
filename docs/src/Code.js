import styled from 'styled-components'
import { color, fontSize } from 'styled-system'

const Code = styled.code`
  font-family: 'Roboto Mono', monospace;
  ${color} ${fontSize};
`

Code.defaultProps = {
  fontSize: 1,
  color: 'darkBlue'
}

export default Code
