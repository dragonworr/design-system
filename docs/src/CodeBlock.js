import React from 'react'
import styled from 'styled-components'
import * as P1 from 'pcln-design-system'
import { LiveProvider, LivePreview, LiveEditor, LiveError } from 'react-live'
import { color, space, borderRadius, theme } from 'styled-system'
import Catch from './Catch'
import Pre from './Pre'

const Preview = styled(LivePreview)`
  padding: 16px;
  border-width: 1px;
  border-style: solid;
  border-color: ${theme('colors.lightGray')};
`

const Err = styled(LiveError)`
  font-family: 'Roboto Mono', Menlo, monospace;
  font-size: 12px;
  top: 0;
  right: 0;
  left: 0;
  padding: 16px;
  color: #fff;
  background-color: #f00;
`

const Editor = styled(LiveEditor)`
  font-family: 'Roboto Mono', monospace;
  font-size: 14px;
  margin: 0;
  padding: 8px;
  tab-size: 2;
  -moz-tab-size: 2;
  -o-tab-size: 2;
  white-space: pre-wrap;
  overflow: auto;
  ${color} &:focus {
    outline: none;
    box-shadow: 0 0 0 1px ${theme('colors.lightBlue')};
  }
`

Editor.defaultProps = {
  color: 'blue',
  bg: 'lightGray'
}

class CodeBlock extends React.Component {
  constructor() {
    super()
    this.getCode = props =>
      React.Children.toArray(props.children)
        .filter(child => typeof child === 'string')
        .join('\n')
  }

  render() {
    const { lang } = this.props

    if (lang !== '.jsx') {
      return <Pre children={this.props.children} />
    }

    const code = this.getCode(this.props)

    return (
      <P1.Box mb={3}>
        <Catch>
          <LiveProvider code={code} scope={P1} mountStylesheet={false}>
            <Preview />
            <Editor />
            <P1.Text
              fontSize="8px"
              align="right"
              caps
              bold
              mt={-12}
              mr={1}
              color="blue"
            >
              Live Code
            </P1.Text>
            <Err />
          </LiveProvider>
        </Catch>
      </P1.Box>
    )
  }
}

export default CodeBlock
