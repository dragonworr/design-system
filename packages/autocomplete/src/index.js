import React from 'react'
import PropTypes from 'prop-types'
import Downshift from 'downshift'
import styled from 'styled-components'
import {
  Flex,
  Label as PclnLabel,
  Input as PclnInput
} from 'pcln-design-system'

export const AutocompleteContext = React.createContext()

export class Autocomplete extends React.Component {
  static Label = Label
  static Input = Input
  static Menu = Menu
  static Item = Item

  static propTypes = {
    match: PropTypes.func
  }

  static defaultProps = {
    match: () => true
  }

  render() {
    const { children, ...props } = this.props

    return (
      <Downshift
        {...props}
        children={state => (
          <div>
            <AutocompleteContext.Provider
              value={{ ...props, ...state }}
              children={children}
            />
          </div>
        )}
      />
    )
  }
}

export const withAutocomplete = (Component, mapProps) =>
  React.forwardRef((props, ref) => (
    <AutocompleteContext.Consumer
      children={state => (
        <Component ref={ref} {...mapProps({ ...props, ...state })} {...props} />
      )}
    />
  ))

export const Label = withAutocomplete(PclnLabel, ({ getLabelProps }) =>
  getLabelProps()
)

export const Input = withAutocomplete(PclnInput, ({ getInputProps }) =>
  getInputProps()
)

export const Menu = ({ children, ...props }) => (
  <AutocompleteContext.Consumer
    children={({ match, isOpen, getMenuProps, inputValue }) =>
      isOpen ? (
        <div
          {...getMenuProps()}
          {...props}
          children={React.Children.toArray(children)
            .filter(el => match(el.props.item, inputValue))
            .map((el, index) => React.cloneElement(el, { index }))}
        />
      ) : (
        false
      )
    }
  />
)

const ItemRoot = styled(Flex)({
  '&[data-selected]': {
    backgroundColor: 'blue'
  },
  '&[data-highlighted]': {
    backgroundColor: 'yellow'
  }
})

export const Item = withAutocomplete(
  ItemRoot,
  ({
    item,
    index,
    getItemProps,
    selectedItem,
    highlightedIndex,
    children,
    ...props
  }) => ({
    px: 2,
    py: 1,
    ...getItemProps({ item, index, ...props }),
    children,
    'data-selected': selectedItem === item ? true : undefined,
    'data-highlighted': highlightedIndex === index ? true : undefined
  })
)
