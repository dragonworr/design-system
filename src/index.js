export { default as Text } from './Text'
export { default as theme } from './theme'
export { default as legacyTheme } from './legacyTheme'
export { default as ThemeProvider } from './ThemeProvider'
export * from './theme'
