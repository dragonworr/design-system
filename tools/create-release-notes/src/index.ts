import { createReleaseNotes } from './actions/createReleaseNotes'

export { createReleaseNotes }
