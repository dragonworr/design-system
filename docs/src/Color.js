import React from 'react'
import styled from 'styled-components'
import { Box, Flex, Heading, Text, theme } from 'pcln-design-system'
import PageTitle from './PageTitle'
import Description from './Description'
import Code from './Code'
import CopyButton from './CopyButton'

const colors = Object.keys(theme.colors)
  .filter(key => !/[0-9]$/.test(key))
  .filter(key => !/^(text|black|white|borderGray)$/.test(key))
  .map(key => ({
    name: key,
    value: theme.colors[key]
  }))
  .filter(color => !Array.isArray(color.value))

const primaries = colors.filter(color => !/^(light|dark)/.test(color.name))
const darks = colors.filter(color => /^dark/.test(color.name))
const lights = colors.filter(color => /^light/.test(color.name))

const ColorChip = styled(Box)`
  height: 0;
  padding-bottom: 75%;
`

const HexButton = styled.button`
  display: inline-block;
  font-family: inherit;
  font-size: 10px;
  margin: 0;
  padding: 0;
  color: inherit;
  background-color: transparent;
  border: 0;
  border-radius: 0;
  appearance: none;
  opacity: 0.75;
  transition-property: opacity;
  transition-duration: 0.125s;
  transition-timing-function: ease-out;

  &:active {
    color: ${props => props.theme.colors.blue};
  }

  & > span {
    opacity: 0;
    transition-property: opacity;
    transition-duration: 0.125s;
    transition-timing-function: ease-out;
  }

  &:hover > span {
    opacity: 1;
  }
`

const ColorCard = ({ name, value }) => (
  <Box>
    <ColorChip bg={value} mb={1} />
    <Text fontSize={0}>{name}</Text>
    <CopyButton text={value}>
      <HexButton>
        <Code fontSize="inherit" color="inherit">
          {value}
        </Code>{' '}
        <span>
          <b>Copy</b>
        </span>
      </HexButton>
    </CopyButton>
  </Box>
)

const Column = props => (
  <Box {...props} width={[1 / 3, null, null, 1 / 6]} mb={3} px={2} />
)

const ColorList = props =>
  props.colors.map(color => (
    <Column key={color.key}>
      <ColorCard {...color} />
    </Column>
  ))

const Color = props => (
  <Box px={3} py={4}>
    <PageTitle>Color</PageTitle>
    <Description>
      The design system includes a color palette of several primary colors,
      along with light and dark variations.
    </Description>
    <Flex wrap mx={-2} py={4}>
      <ColorList colors={primaries} />
      <ColorList colors={lights} />
      <ColorList colors={darks} />
    </Flex>
  </Box>
)

export default Color
