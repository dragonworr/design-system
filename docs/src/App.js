import React from 'react'
import { StaticRouter, BrowserRouter, Route, NavLink } from 'react-router-dom'
import {
  ThemeProvider,
  Flex,
  Box,
  Heading,
  Text,
  Link,
  Hide,
  Container,
  Button
} from 'pcln-design-system'
import Landing from './Landing'
import Color from './Color'
import Typography from './Typography'
import Iconography from './Iconography'
import Detail from './Detail'
import Markdown from './Markdown'
import ScrollTop from './ScrollTop'
import NavItem from './NavItem'
import StickyBar from './StickyBar'

const Router = typeof document !== 'undefined' ? BrowserRouter : StaticRouter

const views = {
  Color,
  Typography,
  Iconography
}

class App extends React.Component {
  constructor() {
    super()

    this.state = {
      menuOpen: false
    }

    this.update = fn => this.setState(fn)
  }

  render() {
    const {
      sections = [],
      content = [],
      basename,
      pathname,
      pkg,
      styles
    } = this.props
    const { menuOpen } = this.state

    return [
      <meta charSet="utf-8" />,
      <title key="title">Priceline One Design System</title>,
      <meta name="viewport" content="width=device-width,initial-scale=1" />,
      // styled-components SSR only returns HTML tag strings or React elements,
      // which isn't ideal here
      <head
        dangerouslySetInnerHTML={{
          __html: styles
        }}
      />,
      <link
        key="webfont"
        rel="stylesheet"
        href="http://fonts.googleapis.com/css?family=Montserrat:400,700|Roboto+Mono"
      />,
      <ThemeProvider key="main">
        <Router basename={basename} location={pathname}>
          <ScrollTop>
            <Flex
              wrap
              align="flex-start"
              color="text"
              style={{ minHeight: '100vh' }}
            >
              <StickyBar
                open={menuOpen}
                width={[1, 256]}
                px={3}
                pt={2}
                pb={5}
                color="white"
                bg="text"
              >
                <Flex align="center" px={2} py={2} mb={3}>
                  <Heading.h1 fontSize={3}>
                    <NavLink
                      to="/"
                      style={{
                        display: 'block',
                        color: 'inherit',
                        textDecoration: 'none'
                      }}
                    >
                      Priceline One
                    </NavLink>
                  </Heading.h1>
                  <Hide ml="auto" sm md lg xl>
                    <Button size="small" onClick={e => this.update(toggleMenu)}>
                      Menu
                    </Button>
                  </Hide>
                </Flex>
                <NavItem to="/" exact>
                  Getting Started
                </NavItem>
                {sections.filter(s => !!s).map(section => (
                  <Box key={section.name} onClick={e => this.update(closeMenu)}>
                    <Heading
                      color="rgba(255, 255, 255, .32)"
                      p={2}
                      bold
                      caps
                      fontSize={0}
                    >
                      {section.name}
                    </Heading>
                    {section.pages.map(page => (
                      <NavItem
                        key={page.name}
                        to={'/' + page.name}
                        color="inherit"
                        children={page.title || page.name}
                      />
                    ))}
                  </Box>
                ))}
                <NavItem href="https://github.com/pricelinelabs/design-system">
                  GitHub
                </NavItem>
              </StickyBar>
              <Box
                width={[1, 'calc(100% - 320px)']}
                style={{
                  flex: '1 1 auto',
                  minHeight: '100vh'
                }}
              >
                <Route
                  exact
                  path="/"
                  render={() => <Landing {...this.props} />}
                />
                <Container maxWidth={768}>
                  {sections.map(section =>
                    section.pages.map((page, i) => (
                      <Route
                        key={page.name}
                        path={'/' + page.name}
                        render={() => {
                          const Component = views[page.name]
                          const pageContent =
                            content.find(c => c.name === page.name) || {}
                          return Component ? (
                            <Component
                              {...page}
                              pages={section.pages}
                              index={i}
                            />
                          ) : (
                            <Detail
                              {...page}
                              content={pageContent.content}
                              pages={section.pages}
                              index={i}
                            />
                          )
                        }}
                      />
                    ))
                  )}
                </Container>
              </Box>
            </Flex>
          </ScrollTop>
        </Router>
      </ThemeProvider>
    ]
  }
}

const pages = [
  {
    name: 'Foundations',
    pages: ['Color', 'Typography', 'Iconography']
  },
  {
    name: 'Components',
    // Component markdown docs can be added here
    pages: [
      'Box',
      'Flex',
      'Hide',
      'Container',
      'Text',
      'Heading',
      'Link',
      'Truncate',
      'Button',
      'Image',
      'BackgroundImage',
      'Icon',
      'Badge',
      'Divider',
      'Card',
      'Hug'
    ]
  },
  {
    name: 'Resources',
    pages: ['Layout', 'Contributing']
  }
]

const customPages = {
  Color: {
    name: 'Color',
    route: '/color'
  },
  Typography: {
    name: 'Typography',
    route: '/typography'
  },
  Iconography: {
    name: 'Iconography',
    route: '/iconography'
  }
}

const routes = pages.reduce((a, b) => [...a, b.pages.map(name => '/' + name)], [
  '/'
])

App.defaultProps = {
  routes
}

// To use a different static site generator, this can be handled in Node
App.getInitialProps = async props => {
  const fs = require('fs')
  const path = require('path')
  const matter = require('gray-matter')
  const { ServerStyleSheet } = require('styled-components')
  const pkg = require('../../package.json')

  const content = fs
    .readdirSync(path.join(__dirname, '..'))
    .filter(file => /\.md$/.test(file))
    .map(filename => {
      const name = path.basename(filename, '.md')
      const raw = fs.readFileSync(path.join(__dirname, '..', filename), 'utf8')
      const { content, data } = matter(raw)

      return Object.assign({}, data, {
        filename,
        name,
        route: '/' + name,
        content
      })
    })

  const sections = pages.map(page => {
    return Object.assign({}, page, {
      pages: page.pages.map(key => {
        return customPages[key] || { name: key, route: '/' + key }
      })
    })
  })

  const sheet = new ServerStyleSheet()
  sheet.collectStyles(React.createElement(props.Component, props))
  const styles = sheet.getStyleTags()

  return {
    pkg,
    styles,
    content,
    sections
  }
}

const toggle = key => state => ({ [key]: !state[key] })
const toggleMenu = toggle('menuOpen')
const closeMenu = state => ({ menuOpen: false })

export default App
