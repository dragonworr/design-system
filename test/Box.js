import React from 'react'
import testRenderer from 'react-test-renderer'
import { Box } from '../src'

describe('Box', () => {
    test('rederer', () => {
      const snapshot = testRenderer.create(<Box />).toJSON()

      expect(snapshot).toMatchSnapshot()
    })
})
