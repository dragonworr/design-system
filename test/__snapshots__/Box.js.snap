// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box rederer 1`] = `
<div
  className="sc-bdVaJa ibTDPh"
/>
`;
